dhis2/dhis2-core
# feat: reset user password with invite [DHIS2-9820]

### Summary
PR adds a new feature allowing a user with proper authority to reset another user login to invitation phase.

Endpoint is:
```
POST /users/{id}/reset
```
(no body needed)

### Automatic Testing
A new test `UserControllerTest` was added that tests the scenarios of the added reset method.
To allow testing the email part a `FakeMessageSender` was added that collects send messages so they can be evaluated in a test. The existing `MockMessageSender` seemed unused so I deleted it.

### Manual Testing
* create a new user, use an email you can check and find out its UID
* switch to superuser
* run `POST /users/{id}/reset` => should give `NO_CONTENT`
* check if you got an email with invitation, follow invitation link and try to complete the "password reset" scenario

Further scenarios:
* try the above scenario but with a user lacking authority and make sure that you cannot reset another user => should be `FORBIDDEN`
* as superuser try reset a non existing user => should be `NOT_FOUND`
* as superuser try reset a user that has no email address set => should be `CONFLICT`
